lib/evmone/rc_ptr.hpp

```diff
@@ -0,0 +1,98 @@
+#include "allocator.hpp"
+
+namespace evmone
+{
+template < class T >
+class rc_ptr
```

Comment:
I think that an `rc_ptr` object can be an 8 byte object, just a pointer. The `make` function will perform just one allocation, allocating memory for both the reference count, pointer to allocator, and `T` in one go. We do not need to have the allocator part of the object if we drop calling `arena->free` in the `release` function, but let us keep it for now.